clarity_ext/domain/analyte.py

```diff
@@ -1,54 +1,54 @@
 from clarity_ext.domain.container import PlatePosition, Well, Container
+from clarity_ext.utils import get_and_apply
 
 
 class Analyte(object):
     """
-    Describes an Analyte in the Clarity LIMS system, including custom UDFs.
+    Describes an Analyte in the Clarity LIMS system.
 
-    Takes an analyte resource as input.
+    Expects certain mappings to UDFs in clarity. These are provided
+    in udf_map, so they can be overridden in different installations.
     """
 
-    def __init__(self, resource, container):
+    def __init__(self, container, name=None, well=None, sample=None, **kwargs):
         """
-        :resource: The API resource
+        Creates an analyte
         """
-        self.resource = resource
         self.container = container
+        self.name = name
+        self.well = well
+        self.sample = sample
 
-    # Mapped properties from the underlying resource
-    @property
-    def name(self):
-        return self.resource.name
+        self.concentration = get_and_apply(kwargs, "concentration", None, float)
+        self.target_concentration = get_and_apply(kwargs, "target_concentration", None, float)
+        self.target_volume = get_and_apply(kwargs, "target_volume", None, float)
 
-    # Mapped UDFs
-    @property
-    def concentration(self):
-        return self.resource.udf["Concentration"]
+    def __repr__(self):
+        return "{} ({})".format(self.name, self.sample.id)
 
-    @property
-    def target_concentration(self):
-        return float(self.resource.udf["Target Concentration"])
+    @staticmethod
+    def create_from_rest_resource(resource, udf_map):
+        container = Container.create_from_rest_resource(resource.location[0], [])
 
-    @property
-    def target_volume(self):
-        return self.resource.udf["Target Volume"]
+        # Map UDFs (which may be using different names in different Clarity setups)
+        # to a key-value list with well-defined key names:
+        analyte_udf_map = udf_map["Analyte"]
+        kwargs = {key: resource.udf.get(analyte_udf_map[key], None) for key in analyte_udf_map}
+        pos = PlatePosition.create(resource.location[1])
+        well = Well(pos, container)
+        sample = Sample.create_from_rest_resource(resource.samples[0])
+        return Analyte(container, resource.name, well, sample, **kwargs)
 
-    # Other properties
-    @property
-    def well(self):
-        pos = PlatePosition.create(self.resource.location[1])
-        return Well(pos, self.container)
 
-    @property
-    def sample(self):
-        return self.resource.samples[0]
+class Sample(object):
+    def __init__(self, sample_id):
+        self.id = sample_id
```

Comment:
Good point. I've changed it, since it always makes sense to send in an ID when creating a sample.

It's difficult to say when one should take parameters as keyword arguments and when not. In this case, I think it's valid that it should be positional.